tests/repository/test_user_repository.py

```diff
@@ -637,3 +637,69 @@ def test_revoke_proposal_permissions_is_idempotent(db_connection: Connection) ->
     # Check that the permission has been revoked
     granted_permissions = user_repository.get_proposal_permissions(user_id)
     assert len(granted_permissions) == 0
+
+
+def test_verify_user_raises_not_found_error_if_not_a_valid_user(db_connection: Connection) -> None:
+    user_repository = UserRepository(db_connection)
+    with pytest.raises(NotFoundError):
+        user_repository.verify_user(-1, False)
+
+
+def _updated_user_verified_status(user_id: int, verify: bool, connection: Connection):
+    user_repository = UserRepository(connection)
+    user_repository.verify_user(user_id, verify)
+    connection.commit()
+    return user_repository.get(user_id)
+
+
+@nodatabase
+def test_verify_user_update_users_verification_status(db_connection: Connection) -> None:
+    username = find_username("Inactive User")
+    user_repository = UserRepository(db_connection)
+    user = user_repository.get_by_username(username)
+
+    # Test user verification is false
+    user = _updated_user_verified_status(user.id, False, db_connection)
+    assert user.user_verified is False
+
+    # Set the verification status to True
+    user = _updated_user_verified_status(user.id, True, db_connection)
+    assert user.user_verified is True
+
+    # Set the verification status back to False
+    user = _updated_user_verified_status(user.id, False, db_connection)
+    assert user.user_verified is False
+
+
+def test_activate_user_raises_not_found_error_if_not_a_valid_user(db_connection: Connection) -> None:
+    user_repository = UserRepository(db_connection)
+    with pytest.raises(NotFoundError):
+        user_repository.activate_user(-1, False)
+
+
+def _get_user_with_updated_active_status(user_id: int, active: bool,  connection: Connection):
```

Comment:
As discussed earlier, the name `_get_user_with_updated_active_status` might be a bit unfortunate.